go/interpreter/lfvm/instructions.go

```diff
@@ -1216,7 +1218,11 @@ func genericCall(c *context, kind tosca.CallKind) {
 	}
 
 	// Get arguments from the memory.
-	args := c.memory.GetSlice(inOffset.Uint64(), inSize.Uint64())
+	args, err := c.memory.readSliceAndExpandMemory(inOffset.Uint64(), inSize.Uint64(), c)
```

Comment:
Could the `expandMemoryWithoutCharging(..)` of line 1195 be removed since the memory expansion is done by this call?

I guess  not, since if the balance check in line 1204 fails, the execution stops without an error but gas for the memory expansion was charged without carrying it out. Best option would be to move this call up line 1195.

Reply:
I would mainly like to keep them apart since the call to `expandMemoryWithoutCharging` is expanding memory to be large enough for both input and output, but this call to `readSliceAndExpand` is only calling with the input size. Yes, this call will never trigger an expansion, but this is ok.